### ISSUE TITLE
Error in creating Participant in MySQL

### ISSUE THREAD

**Issue by phax** [phax/phoss-smp#277]
Hi Philip,
a very similar problem: 
![image](https://github.com/phax/phoss-smp/assets/37209717/6089244f-2086-4c43-8ed0-9f22943e4917)

SMP server version 7.1.1 with SQL backend:
On JDBC settings: amended after checking your post above - no luck:
![image](https://github.com/phax/phoss-smp/assets/37209717/e28097a2-df88-43a7-8e0f-d8929f983c6a)

In logs:
![image](https://github.com/phax/phoss-smp/assets/37209717/4faa58db-297d-45c0-bcff-653b253d15ea)

Database user must have full access to database schema. 
All database tables were created during server startup:
![image](https://github.com/phax/phoss-smp/assets/37209717/b08d87a8-d9c7-41d2-b167-7ccebc9d73b4)

_Originally posted by @skizub in https://github.com/phax/phoss-smp/issues/273#issuecomment-2090894403_

**Comment by phax** [comment]
@skizup lets take this separately.

Can you please show me the "cause" of the SQLException you shown in the logs? Seems some DB specific problem...

**Comment by skizub** [comment]
Philip, yes sure and thanks for moving issue to the right place. 

As all logging available for JDBC is enabled, you can see all SELECTs on the screenshot attached to the original message. For INSERT statement where it seems to be failing to insert into 'smp_service_group' SQL statement is not available so I cannot try it manually to see why process fails. On the other hand it seems to be inserting into 'smp_audit' table ok:
![image](https://github.com/phax/phoss-smp/assets/37209717/71916155-af00-4052-83af-e4ceb2f59e03)
![image](https://github.com/phax/phoss-smp/assets/37209717/1acab2d1-0d5a-448a-b8d4-ef195ad67150)

Yes, forgot to mention - we are using Maria DB v10.5.8, where we moved all our MySQL schemas long time ago without any issues (at least for basic set SQL commands like SELECT, INSERT, UPDATE, DROP).

**Comment by phax** [comment]
@skizub In your original post you added this image:
![327499045-4faa58db-297d-45c0-bcff-653b253d15ea](https://github.com/phax/phoss-smp/assets/196219/4e796b53-0346-44eb-93c1-58f4ed6a1d91)

I would need to know the further details of the stack trace in that image (the "caused by..." stuff). Can you please post it here. Thanks

**Comment by skizub** [comment]
sure

![image](https://github.com/phax/phoss-smp/assets/37209717/3b21e87a-066d-4cf9-99e3-06285e16f107)
![image](https://github.com/phax/phoss-smp/assets/37209717/a1175aa1-6ad8-455c-9fca-83275eb0bc1d)
![image](https://github.com/phax/phoss-smp/assets/37209717/6aba998e-361f-4ccc-be7c-ad3f61d39de1)

**Comment by phax** [comment]
The problem is, that the creation of the Service Group in the SML did not work. Please make sure you selected the correct "SML" in the "SMP Settings". Should be "SMK" for test and "SML" for production.

**Comment by phax** [comment]
Well yes it works now, but also means: Your participants don't end up in the SML and therefore they are not disoverable by other Peppol participants.

Are you using Peppol certificates? If so, Test or Production?

**Comment by skizub** [comment]
That's correct - I just tried to retrieve one of the service groups registered via PEPPOL Practical portal - getting an error saying that service group is not registered. We are using TEST PEPPOL certificates.

**Comment by phax** [comment]
@skizub okay, that is weird. The initial setting is the correct one:

![grafik](https://github.com/phax/phoss-smp/assets/196219/0ebf8c3f-9040-4f40-bff5-b0d382c773eb)


Are you using an outbound proxy server for connections to the Internet?

**Comment by skizub** [comment]
@phax , I'm  not aware of any proxy server used at our end, but I'll try to dig out more details from networking team and get post an update.

**Comment by phax** [comment]
Please check if your configuration contains any of the items mentioned at https://github.com/phax/phoss-smp/wiki/Configuration#httpproxy - the UI should also give an indicator.

But the underlying problem is clearly that you can reach the SML. Maybe it was just a temportary hickup and a retry will work??

**Comment by skizub** [comment]
@phax, thank you - I'll double-check configuration.

Also it doesn't look like a temporary issue, I just tried to set two "SML connection..." fields to **'true'** and got same error back. When I set them back to **'false'**, registration works, but only in local database SML is not being updated. Except proxy option, cannot it be a firewall blocking outgoing connections?

**Comment by phax** [comment]
Yes, it might also be some firewall issue - but they usually prevent connection and don't respond with an HTTP error code. You may try to use the "debug" log level to see the transmission and the headers returned - maybe that helps to see the source of the error.

**Comment by phax** [comment]
Yes thanks - it uses direct connection.
Can you please import your Peppol Test SMP certificate into your browser and access https://acc.edelivery.tech.ec.europa.eu/edelivery-sml/ - it should show a page like this:

![grafik](https://github.com/phax/phoss-smp/assets/196219/251896d3-50e3-4b6f-971f-bf3b1873c4f7)

Or do you also get an HTTP 404 when trying to do so?

**Comment by skizub** [comment]
@phax, tried  to access SML from browser, which I'm using to configure SMP backend. Here is a reply I got:
![image](https://github.com/phax/phoss-smp/assets/37209717/aec8feb1-d9b2-4a96-a2b4-03b9de329786)
Tried to use wget from the server, where backend is running:
 **wget https://acc.edelivery.tech.ec.europa.eu/edelivery-sml/**
![image](https://github.com/phax/phoss-smp/assets/37209717/f53377f2-6c80-4b51-86b4-050bc86bc317)
Also certificates are looking good in SMP frontend:
![image](https://github.com/phax/phoss-smp/assets/37209717/1b12ab79-8b15-4d1a-869f-d4d87083a561)

**Comment by phax** [comment]
Hmm, this is very weird. I am afraid I can't help you any further without looking at the system. Please reach out to me via email (firstname@lastname.com) and lets see if we can have a Zoom session or so...

**Comment by phax** [comment]
We are trying to resolve the topic out of band. It seems to have been a network level configuration issue...

**Comment by skizub** [comment]
@phax , many thanks for your help. All works perfectly now.

Just as an nice to have point for future versions: is it possible to mention that given exception could be linked to the networking issue?

What it says currently somehow leads user to check database...

**Comment by phax** [comment]
Well yes, the base exception is this:

![grafik](https://github.com/phax/phoss-smp/assets/196219/6b8c7428-e278-44e7-9216-778c162bc813)

which "clearly" indicates a miscommunication with the SML server (even though a very odd one).

Also the second level exception indicates that:

![grafik](https://github.com/phax/phoss-smp/assets/196219/be32cb38-e239-4eaa-b5c4-992cf7b93bde)

However I agree, that the error on the UI should highlight that better

**Comment by phax** [comment]
Finally - since the release of DomiSML 5.x the error messages are properly shown :)
